wgengine/userspace.go

```diff
@@ -794,7 +794,9 @@ func (e *userspaceEngine) Reconfig(cfg *wgcfg.Config, routerCfg *router.Config,
 	if err := e.magicConn.SetPrivateKey(wgkey.Private(cfg.PrivateKey)); err != nil {
 		e.logf("wgengine: Reconfig: SetPrivateKey: %v", err)
 	}
-	e.magicConn.UpdatePeers(peerSet)
+	if cfg.Debug != nil && cfg.Debug.RandomizeClientPort {
```

Comment:
this doesn't ever restore it.

how about you set/unset an AtomicBool on magicsock instead and do the `if bool { port = 0 }` later? Then this can just Set(true)  and Set(false)